CDTDatastore/CDTReplicator/CDTPullReplication.m

```diff
@@ -33,7 +34,16 @@ + (instancetype)replicationWithSource:(NSURL *)source target:(CDTDatastore *)tar
 - (instancetype)initWithSource:(NSURL *)source target:(CDTDatastore *)target
 {
     if (self = [super init]) {
-        _source = source;
+        
+        NSURLComponents * sourceComponents = [NSURLComponents componentsWithURL:source resolvingAgainstBaseURL:NO];
```

Comment:
I think I also prefer the small amount of duplication we have currently, and for such a small number of lines I think we can get away with it.